oauthproxy.go

```diff
@@ -151,6 +153,10 @@ func NewOAuthProxy(opts *options.Options, validator func(string) bool) (*OAuthPr
 
 	logger.Printf("Cookie settings: name:%s secure(https):%v httponly:%v expiry:%s domains:%s path:%s samesite:%s refresh:%s", opts.Cookie.Name, opts.Cookie.Secure, opts.Cookie.HTTPOnly, opts.Cookie.Expire, strings.Join(opts.Cookie.Domains, ","), opts.Cookie.Path, opts.Cookie.SameSite, refresh)
 
+	if opts.AuthEndpointAcceptUnauthenticated {
+		logger.Printf("Accepting unauthenticated %s/auth endpoint requests (group based authorization is still active)", opts.ProxyPrefix)
```

Comment:
So this means that if a user is unauthenticated, they are allowed through, but if they are authenticated but have the wrong groups, they aren't allowed through? Is that what we want to happen here? What kind of use cases are you thinking?

Reply:
That is something I couldn’t decide on.
My use case was really for the unauthenticated users (I don’t have a use case for group authorization).
We could render the group authorization useless when the option is enabled and print a log message at startup.

Reply:
This whole concept reminds me of this in Ory Oathkeeper (similar product): https://www.ory.sh/oathkeeper/docs/pipeline/authn#anonymous

If we did anything in this space, I think we should follow a similar pattern:
- X-User header is set explicitly to `anonymous` (and we label flags as similar).
- If any group based authentication is required, we deny the request (`anonymous` user doesn't have the group).

Reply:
So:

- Authorization header if authenticated
- X-User: anonymous header if **un**authenticated
- 403 Forbidden when using group based auth and user is unauthenticated(anonymous)/doesn’t have the group

I could implement this

What name would you like the cfg option to be named?
Currently it is --auth-endpoint-accept-unauthenticated
Should this be changed to --auth-endpoint-accept-anonymous ?

Reply:
Plan sounds good to me

I think because of the nature of the flag, I'd like to prefix it with `insecure`, so maybe `--insecure-auth-endpoint-allow-anonymous-auth`, or is that too much of a mouthful?

Reply:
The flag name makes sense.
I will probably get around to implementing this change next week.